internal/ls/source_map.go

```diff
@@ -18,6 +18,12 @@ func (l *LanguageService) getMappedLocation(fileName string, fileRange core.Text
 		}
 	}
 	endPos := l.tryGetSourcePosition(fileName, core.TextPos(fileRange.End()))
+	if endPos == nil {
+		endPos = &sourcemap.DocumentPosition{
+			FileName: startPos.FileName,
+			Pos:      startPos.Pos + fileRange.Len(),
```

Comment:
Looks right as a port, but since this _is_ happening, do we need to ensure this position is inside the length of the file?

Reply:
Hm, I can. I'm questioning whether or not it's a good idea, though, given we don't do the same sort of bounds checks when going from an LSP position to a byte position... Could go either way. Either way it's sort of papering over something.